### PR TITLE
Assign random colors to Chains in portfolio view.

### PR DESCRIPTION
**Pull request by 0xDaedalus**
Closes https://github.com/tahowallet/extension/issues/3193

Assignment is not quite random - but can be switched to truly random in the future if we deem this is _really_ something we want.

image

### To Test
- Import a wallet with assets across many networks.
- Confirm that each network is assigned a different color.

Latest build: [extension-builds-3204](https://github.com/tahowallet/extension/suites/11946270834/artifacts/626232794) (as of Fri, 31 Mar 2023 14:28:45 GMT).

### CODE REVIEW

**Comment by hyphenized** [comment]
Should colors be assigned randomly for all networks or only custom? cc @VladUXUI

**Comment by 0xDaedalus** [comment]
I did it by modifying the redux store and the component state itself - but you are correct this feature will not see the light of day until we add asset prices on custom networks.

File: ui/components/Overview/NetworksChart.tsx [hunk 1]
```diff
@@ -71,23 +104,23 @@ export default function NetworksChart({
...
         <div className="chains_legend">
-          {percents.map(({ chainID, percent }) => (
+          {percents.map(({ chainID, percent, color }) => (
             <div className="chains_legend_item" key={chainID}>
               <div
                 className="chains_legend_dot"
-                style={{ backgroundColor: NETWORKS_CHART_COLORS[chainID] }}
+                style={{ backgroundColor: color }}
               />
               {NETWORK_BY_CHAIN_ID[chainID].name}({percent}%)
```
Comment (jagodarybacka) [inline ui/components/Overview/NetworksChart.tsx:125]: This line will throw an error as NETWORK_BY_CHAIN_ID includes only built-in networks - we won't know the name of custom network for the chart's legend

**Review by jagodarybacka — CHANGES_REQUESTED** [review]
Found a place where it throws an error:

**Pull request merged**